### PR TITLE
[SYCL][E2E] wrap lit script lines in `%if run-mode`

### PR DESCRIPTION
**Pull request by ayylol**
In #15728 lit script lines that contain `%{run}` `%{run-unfiltered-devices}` or `%if run-mode` are not executed if the system is set to only build e2e tests.

This patch adds `%if run-mode` to lines that require that the test binary has already been executed. This is needed because otherwise these commands are ran on the building system which does not execute the test binaries, creating failures.

### CODE REVIEW

File: sycl/test-e2e/ESIMD/PerformanceTests/matrix_transpose.cpp [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 // RUN: mkdir -p %t.dir && %{build} -o %t.dir/exec.out
 // RUN: env IGC_DumpToCustomDir=%t.dir IGC_ShaderDumpEnable=1 %{run} %t.dir/exec.out
-// RUN: python3 %S/instruction_count.py %t.dir %if igc-dev %{ 1059 %} %else %{ 1116 %} ZTSZZ7runTestjjjRdS_ENKUlRN4sycl3_V17handlerEE_clES3_E3K16.asm
-// RUN: echo "Baseline from driver version 1.3.30872"
+// RUN: %if run-mode %{python3 %S/instruction_count.py %t.dir %if igc-dev %{ 1059 %} %else %{ 1116 %} ZTSZZ7runTestjjjRdS_ENKUlRN4sycl3_V17handlerEE_clES3_E3K16.asm%}
```
Comment (sarnex) [inline sycl/test-e2e/ESIMD/PerformanceTests/matrix_transpose.cpp:13]: if there any way to automatically detect when the command should be run in build or test, or at least get it correct in most cases? having test authors having to add extra code to support a CI optimization they probably don't know about might be burdensome

Comment (ayylol) [reply sycl/test-e2e/ESIMD/PerformanceTests/matrix_transpose.cpp:13]: I did experiment with a couple of heuristics to do this, basically thinking about this in the opposite direction: what lines should only run when building, however i found that there were significantly more exceptions i had to deal with in that case. Doing it in this manner was the way that got it correct in the vast majority of cases (all but the tests included in this pr).

Would be open to ideas for sure though.

Comment (sarnex) [reply sycl/test-e2e/ESIMD/PerformanceTests/matrix_transpose.cpp:13]: oh i see you already handle the standard card of using one of the "{run}" things, so test authors would only have to add this if they were doing something weird, IMO that's fine

we should make sure we tell devs in precommit that its broken in split mode (so i guess we should have to use split build/run in precommit) so they can fix it quickly

**Review by sergey-semenov — APPROVED** [review]

**Comment by ayylol** [comment]
Marking this as a draft until #15728 is further along.

**Pull request closed without merging**